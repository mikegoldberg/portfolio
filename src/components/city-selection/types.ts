export interface CitySelectionProps {
  city: string;
  onSubmit: (name: string) => void;
}
