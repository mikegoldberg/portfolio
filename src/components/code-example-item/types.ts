import { MouseEventHandler } from "react";

export interface CodeExampleItemProps {
  title: string;
  poster: string;
  onClick: MouseEventHandler;
  source: string;
}
