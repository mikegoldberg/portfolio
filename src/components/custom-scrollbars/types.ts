import { ReactElement } from "react";

export interface CustomScrollbarsProps {
  onScrollFrame: (position: number) => void;
  children?: ReactElement;
}
