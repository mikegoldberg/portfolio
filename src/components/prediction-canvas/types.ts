export type PredictionCanvasProps = {
  hide: boolean;
  width: number;
  height: number;
};
