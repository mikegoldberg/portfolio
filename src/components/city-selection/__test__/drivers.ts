export const HEADING = "city-selection-heading";
