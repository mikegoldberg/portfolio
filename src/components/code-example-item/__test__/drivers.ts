export const POSTER = "code-example-item-poster";
export const ONCLICK = "code-example-item-onclick";
export const TITLE = "code-example-item-title";
export const SOURCE = "code-example-item-source";
