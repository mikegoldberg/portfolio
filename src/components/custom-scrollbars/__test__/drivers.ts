export const CONTENT = "custom-scrollbars-content";
