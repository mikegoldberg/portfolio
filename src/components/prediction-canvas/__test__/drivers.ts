export const PREDICTION_CANVAS_CANVAS = "prediction-canvas-canvas";
