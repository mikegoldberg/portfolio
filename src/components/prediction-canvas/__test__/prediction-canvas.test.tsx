import { render } from "../../../test-utils";
import { screen } from "@testing-library/react";
import PredictionCanvas from "..";
import { PredictionCanvasProps } from "../types";
import * as TEST_DRIVERS from "./drivers";

const defaultProps: PredictionCanvasProps = {
  hide: false,
  width: 100,
  height: 100,
};

describe("PredictionCanvas", () => {
  it("renders correctly", () => {
    const { container } = render(<PredictionCanvas {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it("the component is hidden when hide is true", () => {
    render(<PredictionCanvas {...defaultProps} hide={true} />);
    const element = screen.getByTestId(TEST_DRIVERS.PREDICTION_CANVAS_CANVAS);
    const style = getComputedStyle(element);
    const displayValue = style.getPropertyValue("display");
    expect(displayValue).toBe("none");
  });

  it("the component is displayed when hide is false", () => {
    render(<PredictionCanvas {...defaultProps} hide={false} />);
    const element = screen.getByTestId(TEST_DRIVERS.PREDICTION_CANVAS_CANVAS);
    const style = getComputedStyle(element);
    const displayValue = style.getPropertyValue("display");
    expect(displayValue).toBe("block");
  });

  it("the component size is set to the provided width and height", () => {
    render(<PredictionCanvas {...defaultProps} />);
    const element = screen.getByTestId(TEST_DRIVERS.PREDICTION_CANVAS_CANVAS);
    const style = getComputedStyle(element);
    const widthValue = style.getPropertyValue("width");
    const heightValue = style.getPropertyValue("height");
    expect(widthValue).toBe(`${defaultProps.width}px`);
    expect(heightValue).toBe(`${defaultProps.width}px`);
  });
});
