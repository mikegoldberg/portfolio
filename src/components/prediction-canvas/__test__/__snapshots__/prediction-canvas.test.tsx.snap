// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PredictionCanvas renders correctly 1`] = `
<div>
  <canvas
    data-testid="prediction-canvas-canvas"
    style="width: 100px; height: 100px; position: absolute; z-index: 100; display: block;"
  />
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
